code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,156 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	use_power = POWER_USE_OFF
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	var/obj/item/weapon/cell/cell
+	var/on = 0
+	var/drainage = 1
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	..()
+	cell = new/obj/item/weapon/cell/high(src)
```

Comment:
IMO 
```
/obj/machinery/dehumidifier
    var/obj/item/weapon/cell/cell = /obj/item/weapon/cell/high

/obj/machinery/dehumidifier/Initialize()
    if(ispath(cell))
        cell = new cell(src)
    . = ..()
````
Allow for overriding it with different cells later.

Reply:
Done as suggested